pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -604,6 +759,26 @@ func (lbaas *LbaasV2) UpdateLoadBalancer(clusterName string, service *api.Servic
 			if err != nil && !isNotFound(err) {
 				return err
 			}
+			// Delete the rules in the Node Security Group
+			opts := rules.ListOpts{
+				SecGroupID:    lbaas.opts.NodeSecurityGroupID,
+				RemoteGroupID: lbSecGroup.ID,
+				PortRangeMax:  int(port.Port),
+				Protocol:      string(port.Protocol),
+			}
+			rules_, err := getSecurityGroupRules(lbaas.network, opts)
+
+			if err != nil {
```

Comment:
"not found" should not be treated as an error here - it just means we've already partially deleted the loadbalancer, failed, and being retried.